### ISSUE TITLE
More complex navbar menu with dropdowns

### ISSUE THREAD

**Issue by rufuspollock** [flowershow/flowershow#91]
## Job story
I want to have support for a dropdown menu with a max of 1 level

## Acceptance
* [x] onClick&&onHover Menu dropdown  in Desktop view
* [x] onClick Menu dropdown  in Mobile view
* [x] add docs  **added [here](https://github.com/flowershow/flowershow/blob/main/site/content/docs/dropdown.md)**

## Tasks
* [x] Look at what vuepress supports
* [x] Menu dropdown in Desktop view **Menu works onHover and onClick**
* [x] Menu dropdown in Mobile view **Menu works onClick**
* [x] add docs **added [here](https://github.com/flowershow/flowershow/blob/main/site/content/docs/dropdown.md)**

## Analysis
Current idea for implementation is to be able to add dropdown menu items in config.navLinks. If a href prop is not found for an element, then map through it's "subItems" prop array and create dropdown items for each with their respective links by using headless UI.

**Comment by PhilippeduPreez** [comment]
idea for single level implementation
![image](https://user-images.githubusercontent.com/44789132/190255912-82bcbf75-1ac8-468a-b388-9756c25507e3.png)

**Comment by PhilippeduPreez** [comment]
and for multi level
![image](https://user-images.githubusercontent.com/44789132/190256731-2fb92dbc-df9d-431b-b306-d0318dfa6cfd.png)

**Comment by rufuspollock** [comment]
@PhilippeduPreez good start 👏

My suggestion for analysis is to prep this in a hackmd and link that - as easier to edit and comment on and then we can copy across.

I would focus first on a [job story](https://playbook.datopian.com/job-stories/) then on thinking a bit technically how to do it and then finalize in terms of acceptance and tasks.

Aside: small formatting issues in description - markdown for Acceptance and Tasks heading is not working.

**Comment by PhilippeduPreez** [comment]
@rufuspollock Desktop view thus far...

Please let know if you prefer different styling (background hover) etc. Otherwise I use the same styling as the top level nav and will implement the same for mobile view.

Nested dropdowns are a bit more complicated, so please let know if we stick to one level for now or whether we should add nested functionality.

![Animation](https://user-images.githubusercontent.com/44789132/190590436-f00fc275-2549-4c09-a176-fa5a95f0d414.gif)

**Comment by PhilippeduPreez** [comment]
Mobile view thus far

![dropdown mobile](https://user-images.githubusercontent.com/44789132/190608667-19dd536c-73b7-4469-abd9-6fffaaf43dbb.gif)

**Comment by olayway** [comment]
@PhilippeduPreez just to sum up our today's conversation:
Looks great so far! But let's try switching from on-click to on-hover for opening the dropdown (in the navbar) so that the menu item can be a clickable item itself (in our case this could be e.g. docs page, with its guides listed in the dropdown). In the sidebar, it should be on-click, since, as you correctly noticed, there is no hover-over on mobile.
Also, you can already use it for our Docs page, so we have a live example on the Flowershow website :smiley:

**Comment by olayway** [comment]
Another issue is that we wouldn't be able to click on such an item in the sidebar, so we're going with your solution with just a small change - on-click to on-hover for navbar menu items. Once this is finished I'll just rename our docs page to "What is Flowershow", remove links to guides and I'll use all these pages as dropdown items to "Docs" navbar item.

**Comment by PhilippeduPreez** [comment]
FIXED. See commit [here](https://github.com/flowershow/flowershow/pull/177/commits/7c9592977a3f5c02a513ca9f9a7f5b69bf9641e4)

**Comment by rufuspollock** [comment]
I think there is some UI issues atm with how this works:

- Width of sub-items is constrained to width of main item which is problem if text is longer (see image) <img width="399" alt="image" src="https://user-images.githubusercontent.com/180658/200549733-2ee36f3e-83b6-4d47-b301-470e0add61d9.png">
- No indication that subitems exist e.g. no marker. cf current life itself website with the carets or how headlessui does it https://headlessui.com/react/menu <img width="1370" alt="image" src="https://user-images.githubusercontent.com/180658/200549976-ac74b28b-14bb-4c70-82b3-c05d2abca26c.png">
- Hard to separate items (maybe need separator) - may be a consequence of first point

Before we do more on this though we should look at whether we want dropdown navbar at all IMO - see https://github.com/flowershow/flowershow/discussions/256
